### PR TITLE
make logdir timeout configurable

### PR DESCRIPTION
**Pull request by linehrr**
this patch is to make logdir timeout configurable, instead of hard coded value in static. 
we ran into problems when logdir response time is big, so we need to adjust this in prod. 

created new config entry: `logdir.response.timeout.ms` with default the same as before which is 10000 ms. 

On branch configurable_timeouts
 Changes to be committed:
	modified:   cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/KafkaCruiseControlConfig.java
	modified:   cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/Executor.java
	modified:   cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorAdminUtils.java

### CODE REVIEW

**Comment by linehrr** [comment]
@kidkun this PR is the revision of the https://github.com/linkedin/cruise-control/pull/848

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorAdminUtils.java [hunk 1]
```diff
@@ -42,7 +44,8 @@ private ExecutorAdminUtils() {
    * @return Replica logdir information by task.
    */
   static Map<ExecutionTask, ReplicaLogDirInfo> getLogdirInfoForExecutionTask(Collection<ExecutionTask> tasks,
-                                                                              AdminClient adminClient) {
+                                                                             AdminClient adminClient,
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorAdminUtils.java]: Can you update the Javadoc for methods which you add new parameter to?

**Comment by linehrr** [comment]
@kidkun added javadoc, also replaced all places that are using hard coded value. removed that constant now.

**Comment by linehrr** [comment]
hi @kidkun , can you please take a look before conflicts start to show up? I did change quite lot of places in this PR,  thank you.

**Comment by kidkun** [comment]
@linehrr Sorry for the delay again...
Quite busy recently, I will review it today.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java [hunk 2]
```diff
@@ -65,14 +66,23 @@
   private long _executionId;
   private ReplicaMovementStrategy _defaultReplicaMovementTaskStrategy;
   private final AdminClient _adminClient;
+  private final KafkaCruiseControlConfig _config;
 
-  public ExecutionTaskPlanner(AdminClient adminClient, List<String> defaultReplicaMovementStrategies) {
+  /**
+   *
+   * @param adminClient The adminClient to send describeReplicaLogDirs request.
+   * @param defaultReplicaMovementStrategies
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java]: missing a description here.

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorAdminUtils.java [hunk 3]
```diff
@@ -13,6 +13,8 @@
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.TimeoutException;
+
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutorAdminUtils.java]: nit: redundant line

File: cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/MonitorUtils.java [hunk 4]
```diff
@@ -378,14 +379,15 @@ static void setBadBrokerState(ClusterModel clusterModel, Cluster cluster) {
    */
   static Map<TopicPartition, Map<Integer, String>> getReplicaPlacementInfo(ClusterModel clusterModel,
                                                                            Cluster cluster,
-                                                                           AdminClient adminClient) {
+                                                                           AdminClient adminClient,
```
Comment (kidkun) [inline cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/MonitorUtils.java]: can we update Javadoc for this method?

**Comment by linehrr** [comment]
@kidkun thanks for the review, resolved all.

**Pull request merged**